x/incentives/keeper/hooks_test.go

```diff
@@ -352,11 +362,154 @@ func (s *KeeperTestSuite) Test_AfterEpochEnd_Group_ChangeVolumeBetween() {
 	s.validateDistributionForGroup(poolIDsGroup, poolIDToExpectedDistributionMap)
 }
 
-// TODO: create the following tests:
-// https://github.com/osmosis-labs/osmosis/issues/6559
-//
-// Test_AfterEpochEnd_Group_CreateGroupsBetween
-// Test_AfterEpochEnd_Group_SwapAndDistribute
+// This test focuses on a new group being added in-between epochs
+// The structure is:
+// Setup even volume across pools
+// Create Group that is non-perpetual over 2 epochs with 2x defaultCoins
+// Call after epoch hook
+// Update volume to increase but keep being even
+// Create Group that is perpetual and has 1x defaultCoins
+// Call after epoch hook
+// Validate that 3x defaultCoins distributed evenly across pools
+func (s *KeeperTestSuite) Test_AfterEpochEnd_Group_CreateGroupsBetween() {
+	s.SetupTest()
+
+	var volumeA = oneMillionVolumeAmt
+
+	// Create set of pools with internal gauges and a group for them.
+	poolAndGaugeInfo := s.PrepareAllSupportedPools()
+	poolIDsGroup := []uint64{poolAndGaugeInfo.ConcentratedPoolID, poolAndGaugeInfo.StableSwapPoolID}
+	// setup initial volumes so that a Group can be created.
+	s.overwriteVolumes(poolIDsGroup, []osmomath.Int{defaultVolumeAmount, defaultVolumeAmount})
+	// Create non-perpetual group distribution over 2 epochs.
+	_, err := s.App.IncentivesKeeper.CreateGroup(s.Ctx, defaultCoins.Add(defaultCoins...), types.PerpetualNumEpochsPaidOver+2, s.TestAccs[0], poolIDsGroup)
+	s.Require().NoError(err)
+
+	// Setup uneven volumes with volumeA total amount
+	equalPoolVolumes, volumeA := setupEqualVolumeWeights(len(poolIDsGroup), volumeA)
+	poolIDToVolumeMap := map[uint64]osmomath.Int{}
+	s.setupVolumeForPools(poolIDsGroup, equalPoolVolumes, poolIDToVolumeMap)
```

Comment:
This comment should say even not uneven I believe

Reply:
addressed